Eastern-Research-Group/mywaterway
# Feature/community show all waterbodies

## Related Issues:
* I think this goes with https://jira.epa.gov/browse/HMW-5 although the implementation is a little different based on internal and client discussions.

## Main Changes:
* Added a new widget for displaying all waterbodies outside of the selected huc. 
  * This new widget is on by default. 
  * The waterbodies are mostly transparent to differentiate from those waterbodies in the selected huc. 
  * Clicking on a waterbody outside of the huc will result in the same popup as those in the huc, except those outside of the huc have the option to change to the location.
  * The renderer does not update when changing tabs.
* Added a zoom level (zoom = 10 or scale = 577791) to the surrounding waterbodies layer.
  * I chose this value (10) based on the amount of data downloaded. Most areas seem to be around 3MB. Some areas in Pennsylvania download around 15MB. Setting this threshold lower (i.e. 11) seems to make this feature pointless as some larger hucs zoom to 11. 
  * Coincidentally, this zoom threshold matches up with the watersheds zoom threshold.

## Steps To Test:
1. Navigate to http://localhost:3000/community/dc/overview
2. Verify that waterbodies show that are inside and outside of the selected huc
3. Verify that most waterbodies outside of the huc are transparent. This huc has some waterbodies that are solid because they touch the huc
4. Zoom out and verify the surrounding waterbodies layer disappears (coded to disappear at zoom levels less than 10)
5. Click the `Community` tab 
6. Verify the map is blank
7. Verify the `View Surrounding Waterbodies` widget is disabled

## TODO:
- [x] If EPA decides they want to continue with this solution, we need to add the ability for the renderer to be updated when the user changes tabs (i.e., Swimming tab should have different color coding than the Overview tab). 
- [x] Change the icon for the surrounding waterbodies widget.
